lib/internal/url.js

```diff
@@ -932,6 +935,79 @@ function urlToOptions(url) {
   return options;
 }
 
+function getPathFromURLWin32(url) {
+  var hostname = url.hostname;
+  var pathname = url.pathname;
+  if (hostname !== '') {
+    // If hostname is set, then we have a UNC path
+    return `//${hostname}${pathname}`;
+  } else {
+    // Otherwise, it's a local path that requires a drive letter
+    var letter = pathname.codePointAt(1) | 0x20;
+    var sep = pathname[2];
+    if (letter < 96 || letter > 122 ||
+        (sep !== ':' && sep !== '|')) {
+      return new Error('File URLs must specify absolute paths');
+    }
+    pathname = pathname.slice(1);
+    for (var n = 0; n < pathname.length; n++) {
+      if (pathname[n] === '%') {
+        var third = pathname.codePointAt(n + 2) | 0x20;
+        if ((pathname[n + 1] === '2' && third === 102) ||
+            (pathname[n + 1] === '5' && third === 99)) {
+          return new Error('Path must not include encoded \\ or / characters');
+        }
+      }
+    }
+    return decodeURIComponent(pathname);
+  }
+}
+
+function getPathFromURLOther(url) {
```

Comment:
`getPathFromURLPosix` might be a good name, that would mirror `path.win32` and `path.posix`